app/components/primer/popover_component.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module Primer
+  class PopoverComponent < Primer::Component
+    include ViewComponent::Slotable
+
+    with_slot :message, :heading, :body, :button
+
+    class Message < ViewComponent::Slot
+      CARET_DEFAULT = :none
+      CARET_MAPPINGS = {
+        CARET_DEFAULT => "",
+        :bottom => "Popover-message--bottom",
+        :bottom_right => "Popover-message--bottom-right",
+        :bottom_left => "Popover-message--bottom-left",
+        :left => "Popover-message--left",
+        :left_bottom => "Popover-message--left-bottom",
+        :left_top => "Popover-message--left-top",
+        :right => "Popover-message--right",
+        :right_bottom => "Popover-message--right-bottom",
+        :right_top => "Popover-message--right-top",
+        :top_left => "Popover-message--top-left",
+        :top_right => "Popover-message--top-right"
+      }.freeze
+
+      attr_reader :kwargs
+
+      def initialize(caret: CARET_DEFAULT, **kwargs)
+        @kwargs = kwargs
+        @kwargs[:classes] = class_names(
+          kwargs[:classes],
+          "Popover-message text-left p-4 mt-2 mx-auto Box box-shadow-large",
+          CARET_MAPPINGS[fetch_or_fallback(CARET_MAPPINGS.keys, caret, CARET_DEFAULT)]
+        )
+      end
+    end
+
+    class Heading < ViewComponent::Slot
+      attr_reader :kwargs
+
+      def initialize(**kwargs)
+        @kwargs = kwargs
+        @kwargs[:mb] = 2 unless @kwargs.key?(:mb)
```

Comment:
This is me trying to set the default styling I see in examples on https://primer.style/css/components/popover while still allowing the user to override the bottom margin.